src/smc-webapp/codemirror/modes.coffee

```diff
@@ -65,6 +65,7 @@ require('codemirror/mode/verilog/verilog.js')
 require('codemirror/mode/xml/xml.js')
 require('codemirror/mode/xquery/xquery.js')
 require('codemirror/mode/yaml/yaml.js')
+require('codemirror/mode/yaml-frontmatter/yaml-frontmatter.js')
```

Comment:
yaml frontmatter enhances normal markdown to support the yaml introduction. that's used in the cocalc blog, so, that's why it is also enabled for normal markdown. github itself also supports it for the wiki pages. etc.